src/main/kotlin/com/terraformation/backend/accelerator/ReportService.kt

```diff
@@ -0,0 +1,18 @@
+package com.terraformation.backend.accelerator
+
+import com.terraformation.backend.accelerator.db.ReportStore
+import com.terraformation.backend.customer.model.SystemUser
+import com.terraformation.backend.daily.DailyTaskTimeArrivedEvent
+import jakarta.inject.Named
+import org.springframework.context.event.EventListener
+
+@Named
+class ReportService(
+    private val reportStore: ReportStore,
+    private val systemUser: SystemUser,
+) {
+  @EventListener
+  fun on(@Suppress("UNUSED_PARAMETER") event: DailyTaskTimeArrivedEvent) {
+    systemUser.run { reportStore.notifyUpcomingReports() }
```

Comment:
Wrap this in a try/catch that logs the exceptions; we don't want exceptions bubbling out of `@EventListener` methods since that causes handling of the event to be aborted and any subsequent listeners won't be called.